helios-testing/src/main/java/com/spotify/helios/testing/TemporaryJobs.java

```diff
@@ -17,207 +17,100 @@
-import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
-import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.ThreadLocalRandom;
 
 import static com.google.common.base.Charsets.UTF_8;
 import static com.google.common.base.Optional.fromNullable;
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkNotNull;
-import static com.google.common.base.Strings.isNullOrEmpty;
-import static com.google.common.collect.Lists.newArrayList;
-import static com.spotify.helios.testing.Jobs.undeploy;
-import static java.lang.String.format;
-import static java.lang.System.getProperty;
-import static java.util.Arrays.asList;
+import static java.lang.Integer.toHexString;
 import static java.util.concurrent.TimeUnit.MINUTES;
-import static java.util.concurrent.TimeUnit.SECONDS;
 
-public class TemporaryJobs implements TestRule {
+public class TemporaryJobs implements Closeable {
+
   private static final Logger log = LoggerFactory.getLogger(TemporaryJobs.class);
 
-  static final String HELIOS_TESTING_PROFILE = "helios.testing.profile";
+  private static final String HELIOS_TESTING_PROFILE = "helios.testing.profile";
   private static final String HELIOS_TESTING_PROFILES = "helios.testing.profiles.";
-  private static final String DEFAULT_USER = getProperty("user.name");
   private static final Prober DEFAULT_PROBER = new DefaultProber();
-  private static final String DEFAULT_LOCAL_HOST_FILTER = ".+";
-  private static final String DEFAULT_PREFIX_DIRECTORY = "/tmp/helios-temp-jobs";
-  private static final String DEFAULT_TEST_REPORT_DIRECTORY = "target/helios-reports/test";
-  private static final long JOB_HEALTH_CHECK_INTERVAL_MILLIS = SECONDS.toMillis(5);
   private static final long DEFAULT_DEPLOY_TIMEOUT_MILLIS = MINUTES.toMillis(10);
 
+  private final HeliosDeployment heliosDeployment;
   private final HeliosClient client;
   private final Prober prober;
   private final String defaultHostFilter;
-  private final JobPrefixFile jobPrefixFile;
   private final Config config;
   private final Map<String, String> env;
   private final List<TemporaryJob> jobs = Lists.newCopyOnWriteArrayList();
   private final Deployer deployer;
+  private final Undeployer undeployer;
+  private final String jobPrefix;
 
-  private final TemporaryJobReports reports;
-  private final ThreadLocal<TemporaryJobReports.ReportWriter> reportWriter;
-
-  private final ExecutorService executor = MoreExecutors.getExitingExecutorService(
-      (ThreadPoolExecutor) Executors.newFixedThreadPool(
-          1, new ThreadFactoryBuilder()
-              .setNameFormat("helios-test-runner-%d")
-              .setDaemon(true)
-              .build()),
-      0, SECONDS);
+  private static volatile HeliosDeployment soloDeployment;
```

Comment:
IMO it'd make more sense to move the singleton to either a) the `HeliosSoloDeployment` class, or b) it's own class or such.